source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,154 @@
+.. _golang-transactions:
+
+============
+Transactions
+============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use **transactions** with the
+{+driver-long+}. :manual:`Transactions </core/transactions/>` allow you
+to run a series of operations that are not visible from outside the
+transaction until it is committed. If any operation in the
+transaction fails, the driver aborts the transaction and discards all
+data changes before they ever become visible.
+
+In MongoDB, transactions run within logical **sessions**. A
+:manual:`session </reference/server-sessions/>` is a
+grouping of related read or write operations that you want to run
+sequentially. Sessions enable :manual:`causal consistency </core/read-isolation-consistency-recency/#causal-consistency>`
+for a group of operations or allow you to execute operations in an :website:`ACID
+transaction </basics/acid-transactions>`. MongoDB guarantees that the data
+involved in your transaction operations remains consistent, even if the operations
+encounter unexpected errors.
+
+With the {+driver-short+}, you can create a new session from a
+``Client`` instance as a ``Session`` type.
+
+.. warning::
+
+   You should only use ``Session`` with the ``Client`` (or associated
+   ``Database`` or ``Collection``) that created the ``Session``. Using a
+   ``Session`` with a different ``Client`` will result in operation
+   errors.
+
+.. warning::
+   
+   Implementations of ``Session`` are not safe for concurrent use by multiple `goroutines
+   <https://www.golang-book.com/books/intro/10>`__.
```

Comment:
Issue: It might be my own ignorance of Go, but I wonder if this sentence could use more explanation (especially since it's a Warning). I don't really understand it. I'm guessing it means that multiple routines shouldn't access the same `Session` type?

Suggestion: Clarify meaning.

Reply:
goroutines are concurrent go functions. This just means that you should not implement `Session` in multiple concurrent functions. I'm leaning towards keeping this as is because I think users will understand what this warning means if they are accessing more advanced content like this page.